Add ability to jump to buffers

sheoak opened the pull request:
Using ctrl-o you can jump (:drop) to the buffer if it's already open.
Using ctrl-d you can delete the selected buffer(s)

Add the option g:fzf_preview_buffers_jump (default 0) to drop to the
buffer by default. If set to 1 the behavior of ctrl-o will be to open
the buffer in the current window instead.

yuki-yano commented:
CTRL+D is assigned to page down by default in the preview.
If I had to assign it, I think CTRL+X would be appropriate, but what do you think?

Also, it feels inappropriate to pass a bdelete to open_files.
I don't think it's a good idea to put bdelete in the default processor, so I'll put it on hold.

I think the drop is very good.

For future enhancements, I'm thinking of a major refactoring, including the workings of processor.
I plan to use TypeScript and a remote plugin for that.

sheoak commented:
CTRL-x is fine yes!
You want a PR without the bdelete?

(Sorry I was away from my computer for a while)

sheoak committed:
Add ability to jump to buffers

Using ctrl-o you can jump (:drop) to the buffer if it's already open.

Add the option g:fzf_preview_buffers_jump (default 0) to drop to the
buffer by default. If set to 1 the behavior of ctrl-o will be to open
the buffer in the current window instead.

sheoak commented:
@yuki-ycino I have removed the bdelete/ctrl-d as discussed.

yuki-yano committed:
Add g:fzf_preview_buffers_jump default value

yuki-yano committed:
Add g:fzf_preview_drop_key_map document

yuki-yano commented:
I've fixed this PR issue and added a similar implementation to the TypeScript plugin I'm currently working on.
[`29fde07` (#133)](https://github.com/yuki-ycino/fzf-preview.vim/pull/133/commits/29fde07a7a5e867c370552a138e6fe982831ffaf)

This PR will be merged. Thanks.

yuki-yano commented:
TypeScript allows for flexibility in design, so it's likely to be possible to implement the removal of buffers.
Ctrl-x is used for splitting and Ctrl-d is used for preview scrolling, so it can't use.

I think Ctrl-k would be better, but what do you think?

sheoak commented:
Maybe something with an easier mnemonic, not sure what is available, maybe ctrl-backspace?

yuki-yano commented:
@sheoak Added a mechanism for TypeScript to handle processes flexibly, and changed Buffers and AllBuffers to remove buffers with C-q.
https://github.com/yuki-ycino/fzf-preview.vim/commit/88633bf00ca7d32625fbc3909e6161bc98f4f5d5
https://github.com/yuki-ycino/fzf-preview.vim/commit/d537a5d50b51862541276e22752f7e5bc7395113
